bindings/java/src/main/java/org/ucx/jucx/ucp/UcpWorkerParams.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2019. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+package org.ucx.jucx.ucp;
+
+import java.nio.ByteBuffer;
+import java.util.BitSet;
+
+import org.ucx.jucx.UcxParams;
+import org.ucx.jucx.ucs.UcsConstants;
+
+public class UcpWorkerParams extends UcxParams {
+
+    private int threadMode;
+
+    private BitSet cpuMask = new BitSet();
+
+    private long events;
+
+    private ByteBuffer userData;
+
+    private int eventFD;
+
+    @Override
+    public UcpWorkerParams clear() {
+        super.clear();
+        threadMode = 0;
+        cpuMask = new BitSet();
+        events = 0;
+        userData = null;
+        eventFD = 0;
+        return this;
+    }
+
+    /**
+     * Requests the thread safety mode which worker and the associated resources
+     * should be created with. By default UCS_THREAD_MODE_SINGLE is used.
+     * When thread safety requested, the {@link UcpWorker#UcpWorker(UcpContext)}
+     * attempts to create worker where multiple threads can access concurrently.
+     * The thread mode with which worker is created can differ from the
+     * suggested mode.
```

Comment:
pls add a link to ucp_worker_query (but in java), this is an API how to check real state.

Reply:
Propose to add in a separate PR. Since this PR is already 500 lines.

Reply:
Ok